Docs: Add docs for Spark SQL Iceberg transform functions (#13156)

VerneZhong opened the pull request:
This PR adds documentation for the following Iceberg-related SQL functions:
	•	system.iceberg_version
	•	system.bucket
	•	system.years
	•	system.month
	•	system.days
	•	system.hours
	•	system.truncate

These functions are used in Iceberg partition transforms. The docs include descriptions, argument types, return values, and examples.

VerneZhong commented:
@manuzhang Hi, this PR adds documentation for Iceberg SQL functions. Please help review when you have time. Also, the workflow is pending your approval. Thanks!

VerneZhong commented:
@manuzhang
Added missing license header to fix CI failure.

manuzhang commented:
cc @nastra @RussellSpitzer please help review.

nastra commented:
@VerneZhong can you please rebase the PR so that it doesn't contain 200 commits but only a single one with the changes that you're proposing?

VerneZhong commented:
@nastra Sure! I’ll rebase the PR and squash it into a single commit as requested.
Thanks for the heads-up!

nastra commented:
can you please deploy the site locally as described in https://github.com/apache/iceberg/blob/a8d111eaa7bfb3f98a236578cee3b2ff14b7b338/site/README.md and share a screenshot? This is to make sure that the site is properly formatted so that we don't have to do follow-up PRs to fix the indentation and such

VerneZhong commented:
Hi @nastra , I’ve addressed the review comments and updated the PR accordingly. I also deployed the site locally as described in the [README](https://github.com/apache/iceberg/blob/a8d111eaa7bfb3f98a236578cee3b2ff14b7b338/site/README.md), and everything looks properly formatted. I’ve attached a screenshot for your reference. Please let me know if anything else is needed!

<img width="1510" height="747" alt="image" src="https://github.com/user-attachments/assets/6af0250a-ed19-4a36-8ea2-6f7a6fbf39fa" />

nastra commented on docs/docs/spark-functions.md:
```diff
@@ -0,0 +1,262 @@
+
+---
+
+### `system.bucket(value, num_buckets)`
```
parameters are in the wrong order

nastra commented on docs/docs/spark-functions.md:
````diff
@@ -0,0 +1,262 @@
+SELECT system.bucket(column_value, num_buckets);
+```
+
+#### Parameters
````
there's also `Arguments` a few lines above

nastra commented on docs/docs/spark-functions.md:
```diff
@@ -0,0 +1,262 @@
+SELECT system.months(DATE '2017-12-01');
+-- Result: 575
+
+SELECT system.months(TIMESTAMP '1969-12-31 23:59:59');
```
maybe also add an example with 1970, similarly to https://github.com/apache/iceberg/blob/6c98a9e470d44f77e1939a1a333bf197a57949a6/spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkMonthsFunction.java#L45

nastra commented on docs/docs/spark-functions.md:
```diff
@@ -0,0 +1,262 @@
+SELECT system.days(DATE '2017-12-01');
+-- Result: 17596
+
+SELECT system.days(TIMESTAMP '1969-12-31 23:59:59');
```
same as above. An example with 1970 as shown in TestSparkDaysFunction would be good here

VerneZhong commented:
@nastra Thanks for the review! I’ve addressed all the comments. Please take another look when you have time.
